packages/prompts/src/index.ts

```diff
@@ -1014,3 +1014,482 @@ export const tasks = async (tasks: Task[], opts?: CommonOptions) => {
+				if (opts.initialValue !== undefined && !selectedValue) {
+					const initialIndex = opts.options.findIndex(o => o.value === opts.initialValue);
+					if (initialIndex !== -1) {
+						cursor = initialIndex;
+						selectedValue = opts.options[initialIndex].value;
+					}
+				}
+			}
+			
+			// Set selection on submit
+			if (this.state === 'submit') {
+				this.value = selectedValue as any;
+			}
+			
+			// Title and message display
+			const title = `${color.gray(S_BAR)}\n${symbol(this.state)}  ${opts.message}\n`;
+			
+			// Handle different states
+			switch (this.state) {
+				case 'submit': {
+					// Show selected value
+					const selected = opts.options.find(o => o.value === selectedValue);
+					const label = selected ? getLabel(selected) : '';
+					return `${title}${color.gray(S_BAR)}  ${color.dim(label)}`;
+				}
+				
+				case 'cancel': {
+					return `${title}${color.gray(S_BAR)}  ${color.strikethrough(color.dim(this.value || ''))}`;
+				}
+				
+				default: {
+					// Mode indicator (for debugging)
+					const modeIndicator = isNavigating 
+						? color.yellow(' [navigation]') 
+						: '';
+						
+					// Display cursor position - show plain text in navigation mode
+					const searchText = isNavigating 
+						? color.dim(input)
+						: (this.value ? this.valueWithCursor : color.inverse(color.hidden('_')));
+					
+					// Show match count if filtered
+					const matches = filtered.length !== opts.options.length
+						? color.dim(` (${filtered.length} match${filtered.length === 1 ? '' : 'es'})`)
+						: '';
+					
+					// Render options with selection
+					const displayOptions = filtered.length === 0 
+						? [] 
+						: limitOptions({
+							cursor: cursor,
+							options: filtered,
+							style: (option, active) => {
+								const label = getLabel(option);
+								const hint = option.hint ? color.dim(` (${option.hint})`) : '';
+								return active 
+									? `${color.green(S_RADIO_ACTIVE)} ${label}${hint}` 
+									: `${color.dim(S_RADIO_INACTIVE)} ${color.dim(label)}${hint}`;
+							},
+							maxItems: opts.maxItems,
+							output: opts.output
+						});
+					
+					// Show instructions
+					const instructions = isNavigating
+						? [`${color.dim('↑/↓')} to select, ${color.dim('Enter:')} confirm, ${color.dim('Type:')} to search`]
+						: [`${color.dim('Type')} to filter, ${color.dim('↑/↓')} to navigate`];
+					
+					// No matches message
+					const noResults = filtered.length === 0 && input
+						? [`${color.cyan(S_BAR)}  ${color.yellow('No matches found')}`]
+						: [];
+					
+					// Return the formatted prompt
+					return [
+						title,
+						`${color.cyan(S_BAR)}  ${color.dim('Search:')} ${searchText}${matches}${modeIndicator}`,
+						...noResults,
+						...displayOptions.map(option => `${color.cyan(S_BAR)}  ${option}`),
+						`${color.cyan(S_BAR)}  ${color.dim(instructions.join(' • '))}`,
+						`${color.cyan(S_BAR_END)}`
+					].join('\n');
```

Comment:
The variable `title` already end with a `\n`, so by joining all lines with another `\n` we have an empty line between the title and the rest of the prompt

Either the `\n` need to be removed in the title variable (and then re-added everywhere) or it needs to be trim just here.

```suggestion
					return [
						title.replace(/\n$/, ''),
						`${color.cyan(S_BAR)}  ${color.dim('Search:')} ${searchText}${matches}${modeIndicator}`,
						...noResults,
						...displayOptions.map(option => `${color.cyan(S_BAR)}  ${option}`),
						`${color.cyan(S_BAR)}  ${color.dim(instructions.join(' • '))}`,
						`${color.cyan(S_BAR_END)}`
					].join('\n');
```